### PR TITLE
add filter support in reports

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

**Comment by pirkogdc** [comment]
code looks good

File: src/main/java/com/gooddata/md/report/Filter.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Filter (in report definition)
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+public class Filter {
+
+    private final String expression;
+
+    @JsonCreator
+    public Filter(@JsonProperty("expression") final String expression) {
```
Comment (martiner) [inline src/main/java/com/gooddata/md/report/Filter.java]: is this class going to be created by the user?

Comment (sadam21) [inline src/main/java/com/gooddata/md/report/Filter.java]: Yes, user can create report containing some filters.

Comment (martiner) [inline src/main/java/com/gooddata/md/report/Filter.java]: May the expression be null or empty?

Comment (sadam21) [inline src/main/java/com/gooddata/md/report/Filter.java]: Not empty check added.

**Pull request merged**